src/hypothesis/internal/compat.py

```diff
@@ -501,11 +501,17 @@ def implements_iterator(it):
 
 
 if PY2:
+    # We use integer operations because math.floor etc return floats, which
+    # lose odd numbers at large values.  See tests/py2/test_ceil_floor.py
```

Comment:
minor: I’m not a super fan of having comments like this refer across to tests, because:

1. files change and move around
2. it’s clearer just to have a sentence here, rather than go digging for it in test output

So I’d rather have a comment explaining the exact failure case here, with an example.

Also, is “odd” in this sense meaning “strange” or “not evenly divisible by 2”? I’d be more surprised if it was the latter, but I want to be clear.